buildtest/utils/file.py

```diff
@@ -143,3 +143,66 @@ def resolve_path(path):
 
     if os.path.exists(real_path):
         return real_path
+
+
+def read_file(filepath):
+    """ This method provides capability to read a file.
+
+        Parameters:
+
+        :param filepath: file name to read
+        :type filepath: str, required
+        :return: return content of file
+        :rtype: list
+    """
+
+    # type check filepath to ensure its a string, if not return None
+    if not isinstance(filepath, str):
+        return None
```

Comment:
You have the return type here a list? I would also add a note that returning None indicates some problem with the file. Is there any reason not to exit on error given the return is None? We want to make sure that any Nones returned are properly handled.

Reply:
this is checking is to ensure ``filepath``   is of type string. I could with System Error on this if that makes sense.

Reply:
That makes sense, yes.

Reply:
this is resolved but with a SystemError message see https://github.com/shahzebsiddiqui/buildtest-framework/commit/e4ad6f2188b00e085e92d94b225b8587e1c03cdf